build.sbt

```diff
@@ -5,139 +5,188 @@ import com.typesafe.sbt.web.SbtWeb.autoImport._
 import com.gu.Dependencies._
 import com.gu.ProjectSettings._
 
-val common = library("common").settings(
-  javaOptions in Test += "-Dconfig.file=common/conf/test.conf",
-  libraryDependencies ++= Seq(
-    guBox,
-    apacheCommonsMath3,
-    awsCore,
-    awsCloudwatch,
-    awsDynamodb,
-    awsS3,
-    awsSns,
-    awsSts,
-    awsSqs,
-    awsSsm,
-    contentApiClient,
-    enumeratumPlayJson,
-    filters,
-    commonsLang,
-    jodaConvert,
-    jodaTime,
-    jSoup,
-    liftJson,
-    json4s,
-    playGoogleAuth,
-    quartzScheduler,
-    redisClient,
-    rome,
-    romeModules,
-    scalaCheck,
-    nScalaTime,
-    ws,
-    faciaFapiScalaClient,
-    closureCompiler,
-    jerseyCore,
-    jerseyClient,
-    cssParser,
-    w3cSac,
-    logback,
-    kinesisLogbackAppender,
-    targetingClient,
-    scanamo,
-    scalaUri,
-    commercialShared,
-    playJson,
-    playJsonExtensions,
-    playJsonJoda,
-    jodaForms,
-    jacksonDataFormat,
-    atomRenderer,
-    identityModel,
-    capiAws,
-    okhttp,
-    jsonSchema
-  ) ++ jackson
-).settings(
-    mappings in TestAssets ~= filterAssets
+val sharedSettings = Seq(
+  (test in Test) := (test in Test).dependsOn(scalafmtCheckAll).value
```

Comment:
have you tried
`ThisBuild / Test / test := (Test / test).dependsOn(scalafmtCheckAll).value`
Not totally sure whether it wouldn't be overridden at project level, but try it, that might save having to do it individually on each project.